plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -419,18 +419,19 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
       };
    
       // The queue storing read-only transactions to be executed by read-only threads
+      // Thread-safe
       class ro_trx_queue_t {
       public:
          void push_back(ro_trx_t&& trx) {
             std::unique_lock<std::mutex> g( mtx );
             queue.push_back(std::move(trx));
-            notify_waiting(false);
+            cond.notify_one();
```

Comment:
```suggestion
            if (num_waiting)
               cond.notify_one();
```

Reply:
Maybe we don't want to notify while in write window. `if (num_waiting)` can cover that.